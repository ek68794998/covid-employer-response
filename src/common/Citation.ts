import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
