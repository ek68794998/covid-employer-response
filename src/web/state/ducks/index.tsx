import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});