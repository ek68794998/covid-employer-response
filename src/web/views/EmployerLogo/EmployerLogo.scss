@import "../Colors.scss";
@import "../Dimensions.scss";

$body-color: $outline-color;
$border-radius: 12px;
$card-shadow-color: rgba(0, 0, 0, 0.2);
$icon-size: 32px;
$padding-interior: 12px;
$padding-sides: 16px;

.EmployerListItem__Container {
	background: $background;
	border: 1px solid $body-color;
	border-radius: $border-radius;
	box-shadow: 0 0 0 $card-shadow-color;
	display: flex;
	flex-direction: column;
	font-size: 16px;
	padding: $padding-interior $padding-sides;
	transition: 0.1s all;
	width: 100%;

	&:hover {
		box-shadow: 0px 3px 6px $card-shadow-color;
		transform: translateY(-1px);

		.EmployerDetailsHeader__AggregateRatings {
			opacity: 1.0;
		}

		a.EmployerActionLinks__Link {
			opacity: 1.0;
		}
	}

	.EmployerDetailsHeader__Title h2 {
		flex: 1;
	}

	.EmployerDetailsHeader__TitleGap {
		display: none;
	}
}

.EmployerListItem__Summary {
	cursor: pointer;
	flex: 1;
	font-size: 14px;
	line-height: 180%;
	overflow-y: hidden;
	position: relative;

	p {
		margin: 0;
	}

	.EmployerListItem__OverflowScreen {
		position: absolute;
		bottom: 0;
		left: 0;
		right: 0;
		height: 30px;
		background: linear-gradient(to top, $background, rgba(255, 255, 255, 0));
	}
}

.EmployerListItem__Actions {
	display: flex;
	font-family: 'Baloo Chettan 2';
	justify-content: flex-end;
	margin: 3px 0 0;

	a.EmployerListItem__ReadMore {
		align-items: center;
		display: flex;
		flex: 1;
		justify-content: flex-end;
		margin-right: 0;

		i {
			margin-left: 4px;
		}
	}
}
