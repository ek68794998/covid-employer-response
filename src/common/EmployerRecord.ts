import { Citation } from "./Citation";
import { EmployerLocation } from "./EmployerLocation";
import { EmployerRating } from "./EmployerRating";

export class EmployerRecord {
	public citations: Citation[] = [];

	public employeesBeforeMax: number = -1;

	public employeesBeforeMin: number = -1;

	public id: string = "";

	public image?: string;

	public location: EmployerLocation = new EmployerLocation();

	public name: string = "";

	public officialWebsite?: string;

	public rating: EmployerRating = "fair";

	public summary: string = "";

	public wiki?: string;
}
