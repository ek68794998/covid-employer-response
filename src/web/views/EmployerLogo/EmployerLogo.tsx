import React from "react";
import { RouteProps } from "react-router-dom";

import { EmployerRecordBase } from "../../../common/EmployerRecordBase";

import "./EmployerLogo.scss";

interface Props extends RouteProps {
	employer: EmployerRecordBase;
}

const EmployerLogo: React.FC<Props> = (props: Props): React.ReactElement | null => {
	const { employer } = props;

	const logoImageRegex: RegExpExecArray | null =
		employer.image ? EmployerRecordBase.IMAGE_REGEX.exec(employer.image) : null;

	if (!logoImageRegex) {
		return null;
	}

	return (
		<img
			className="EmployerLogo__Image"
			src={`/images/employers/${logoImageRegex[1]}`}
			style={{ background: logoImageRegex[2] || "#fff" }}
		/>
	);
};

export default EmployerLogo;
