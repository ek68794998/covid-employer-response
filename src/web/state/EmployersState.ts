import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}