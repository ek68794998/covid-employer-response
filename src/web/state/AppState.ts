import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}