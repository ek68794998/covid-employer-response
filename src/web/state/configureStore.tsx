import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;