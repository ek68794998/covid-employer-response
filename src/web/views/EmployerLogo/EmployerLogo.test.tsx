import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { EmployerRecord } from "../../../common/EmployerRecord";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerLogo from "./EmployerLogo";

describe("<EmployerLogo />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerLogo employer={new EmployerRecord()} />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
