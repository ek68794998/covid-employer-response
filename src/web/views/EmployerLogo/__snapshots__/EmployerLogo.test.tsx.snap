// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerLogo /> renders without exploding 1`] = `
<img
  className="EmployerLogo__Image"
  src="/images/employers/test.svg"
  style={
    Object {
      "background": "#123",
    }
  }
/>
`;
